server/src/test/java/org/elasticsearch/transport/TransportServiceHandshakeTests.java

```diff
@@ -61,23 +67,36 @@ public static void startThreadPool() {
     private List<TransportService> transportServices = new ArrayList<>();
 
     private NetworkHandle startServices(String nodeNameAndId, Settings settings, Version version) {
-        MockTcpTransport transport =
-                new MockTcpTransport(settings,
-                                     threadPool,
-                                     PageCacheRecycler.NON_RECYCLING_INSTANCE,
-                                     BigArrays.NON_RECYCLING_INSTANCE,
-                                     new NoneCircuitBreakerService(),
-                                     new NamedWriteableRegistry(Collections.emptyList()),
-                                     new NetworkService(Collections.emptyList())
-                                     );
-        TransportService transportService = new MockTransportService(settings, transport, threadPool,
-                                                                     (boundAddress) -> new DiscoveryNode(
-            nodeNameAndId,
-            nodeNameAndId,
-            boundAddress.publishAddress(),
-            emptyMap(),
-            emptySet(),
-            version), null);
+        var allSettings = Settings.builder()
+            .put(TransportSettings.PORT.getKey(), ESTestCase.getPortRange())
+            .put(settings)
+            .build();
+        var transport = new Netty4Transport(
+            allSettings,
+            Version.CURRENT,
```

Comment:
Using `Version.CURRENT` here instead of `version` is kinda wrong, but it is what the `MockTcpTransport` defaulted too, and without it the `testIncompatibleVersions` test case breaks - because it will raise an exception in the `InboundDecoder` (which triggers onCaughtException in the netty handler, which has a call to `ExceptionsHelper.maybeDieOnAnotherThread(cause);` - so we can't really catch these exceptions in the test.

Because of all that I kept the test as is - even though it is kinda broken.

Reply:
Maybe worth adding your description as a comment here?

Reply:
:+1: 

Done